Fix window title of wallet loading window

ShaMan239 opened the pull request:
The title of the window while loading of wallet is “Bitcoin-Qt”, whereas it should be “Bitcoin Core”. This PR fixes this difference.

Changes introduced in this PR (Runned Bitcoin-GUI on signet network)
|Master|PR|
|---|---|
|![Screenshot from 2021-08-24 00-02-18](https://user-images.githubusercontent.com/85434418/130500309-2f0af2c9-55f0-4609-a92b-3156800fa92e.png)|![Screenshot from 2021-08-23 23-58-26](https://user-images.githubusercontent.com/85434418/130500351-546ee3ae-ea34-4baf-92f7-d9c2012e750d.png)|

MarcoFalke commented:
This will also break the settings in the registry, IIRC

hebasto commented:
Considering the following

https://github.com/bitcoin-core/gui/blob/d3203a99d886177eee9d1f9cd8411e215118a4e6/src/qt/bitcoin.cpp#L516

user could end with two installed versions of Bitcoin Core, no?

hebasto commented:
Approach NACK.

Why not modify `WalletControllerActivity::showProgressDialog`?

Or use [`QGuiApplication::applicationDisplayName`](https://doc.qt.io/qt-5/qguiapplication.html#applicationDisplayName-prop)? But not sure about the latter...

jarolrod reviewed (commented):
This is quite an invasive change, why not just do:

``` diff
diff --git a/src/qt/walletcontroller.cpp b/src/qt/walletcontroller.cpp
index 3cceb5ca5..875dd0c14 100644
--- a/src/qt/walletcontroller.cpp
+++ b/src/qt/walletcontroller.cpp
@@ -206,6 +206,7 @@ void WalletControllerActivity::showProgressDialog(const QString& label_text)
     m_progress_dialog->setLabelText(label_text);
     m_progress_dialog->setRange(0, 0);
     m_progress_dialog->setCancelButton(nullptr);
+    m_progress_dialog->setWindowTitle(PACKAGE_NAME);
     m_progress_dialog->setWindowModality(Qt::ApplicationModal);
     GUIUtil::PolishProgressDialog(m_progress_dialog);
     // The setValue call forces QProgressDialog to start the internal duration estimation.
```

ShaMan239 commented:
Updated  from bb7aae1 to abcc83a ([pr409.01](https://github.com/ShaMan239/gui/tree/pr409.01) -> [pr409.02](https://github.com/ShaMan239/gui/tree/pr409.02))
Addressed Comments by @hebasto and @jarolrod.

Changes made:
- Resetted QAPP_APP_NAME variables to their original value
- Changed title of loading window, by updating WalletControllerActivity::showProgressDialog, as suggested by @hebasto and @jarolrod
- Updated PR description accordingly.

promag commented on src/qt/walletcontroller.cpp:
```diff
@@ -203,6 +203,7 @@ void WalletControllerActivity::showProgressDialog(const QString& label_text)
     assert(!m_progress_dialog);
     m_progress_dialog = new QProgressDialog(m_parent_widget);
 
+    m_progress_dialog->setWindowTitle(PACKAGE_NAME);
```
Search for `setWindowTitle`, typically windows/dialogs use titles. I'd add a `title` argument to `showProgressDialog` and update all call sites, like:
```cpp
showProgressDialog(tr("Open Wallet"), tr("Opening Wallet <b>%1</b>…").arg(name.toHtmlEscaped()));
```
which uses same text as the menu item.

ShaMan239 replied:
Addressed in  58dca47. Thanks for the suggestion @promag.

promag reviewed (commented):
Concept ACK.

ShaMan239 commented:
Updated from abcc83a to 58dca474673f190e947db4bd350926f1a3bd592c ([pr409.02](https://github.com/ShaMan239/gui/tree/pr409.02) -> [pr409.03](https://github.com/ShaMan239/gui/tree/pr409.03))
Addressed Comments by @hebasto and @promag. Thanks, @hebasto, for pointing out the problem, and thanks, @promag, for the suggestion.

Changes made:
- Updated the `showProgressDialog` function to accept two arguments.
- Updated all the occurrences of `showProgressDialog` with the new title as “Open Wallet.”

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -251,7 +252,7 @@ void CreateWalletActivity::askPassphrase()
 void CreateWalletActivity::createWallet()
 {
-    showProgressDialog(tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
+    showProgressDialog(tr("Open Wallet"), tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
```
`Create Wallet` is more appropriate than `Open Wallet` here, no?
```suggestion
    showProgressDialog(tr("Create Wallet"), tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
```

ShaMan239 replied:
Addressed in  d9e421c. Thanks for pointing that out.

ShaMan239 commented:
Updated from 58dca47 to d9e421c3ba937fd37529a74dff07dcc6ec1cdc26 ([pr409.03](https://github.com/ShaMan239/gui/tree/pr409.03) -> [pr409.04](https://github.com/ShaMan239/gui/tree/pr409.04))
Addressed @jarolrod comment.

Changes:
- Changed title of Create Wallet loading window from “Open Wallet” to “Create Wallet.”

hebasto commented on src/qt/walletcontroller.cpp:
```diff
@@ -251,7 +252,7 @@ void CreateWalletActivity::askPassphrase()
 void CreateWalletActivity::createWallet()
 {
-    showProgressDialog(tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
+    showProgressDialog(tr("Create Wallet"), tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
```
nit
> ... which uses the same text as the menu item.

Maybe mention this in the translator comment?

hebasto reviewed (approved):
ACK d9e421c3ba937fd37529a74dff07dcc6ec1cdc26

In commit message maybe
s/"qt: Fix window title of wallet loading window"/"qt: Fix WalletControllerActivity progress dialog title"/
?

shaavan commented:
Updated from d9e421c to 8e11d89 ([pr409.04](https://github.com/shaavan/gui/tree/pr409.04) -> [pr409.05](https://github.com/shaavan/gui/tree/pr409.05))
Addressed @hebasto suggestions

Changes:
- Updated commit message to better explain the intent and changes done in this PR.
- Added Translator comments. I would request reviewers to review the comments and provide their valuable suggestions.

hebasto reviewed (approved):
ACK 8e11d89ea90be5433905ed207ff9a5a5dbe3e8cb

shaavan commented:
Updated from 8e11d89ea90be5433905ed207ff9a5a5dbe3e8cb to e4bf5804bdeee2b16d5caea64b97f002d8d625e5 ([pr409.05](https://github.com/shaavan/gui/tree/pr409.05) -> [pr409.06](https://github.com/shaavan/gui/tree/pr409.06))

Updates:
- Rebased on current Master

hebasto commented:
> Updated from [8e11d89](https://github.com/bitcoin-core/gui/commit/8e11d89ea90be5433905ed207ff9a5a5dbe3e8cb) to [e4bf580](https://github.com/bitcoin-core/gui/commit/e4bf5804bdeee2b16d5caea64b97f002d8d625e5) ([pr409.05](https://github.com/shaavan/gui/tree/pr409.05) -> [pr409.06](https://github.com/shaavan/gui/tree/pr409.06))
> 
> Updates:
> 
>     * Rebased on current Master

It seems wrong rebase...

shaavan commented:
Updated from e4bf5804bdeee2b16d5caea64b97f002d8d625e5 to a086a3757b1edb06b601457142d36628ae15441d ([pr409.06](https://github.com/shaavan/gui/tree/pr409.06) -> [pr409.07](https://github.com/shaavan/gui/tree/pr409.07))

Changes:
- Corrected the rebasing of PR on the current Master.
- Used “Load Wallet” as the title for the progress window, which is displayed when wallets are being loaded at the startup of GUI.
- Added translator’s comment for the “Load Wallet” progress window accordingly.

shaavan commented:
@hebasto @jarolrod 
Would you mind reviewing the PR again after the recent commit? Thanks

hebasto commented on src/qt/walletcontroller.cpp:
```diff
@@ -340,7 +349,11 @@ LoadWalletsActivity::LoadWalletsActivity(WalletController* wallet_controller, QW
 {
-    showProgressDialog(tr("Loading wallets…"));
+    showProgressDialog(
+      //: Title of progress window when which is displayed when wallets are being loaded.
+      tr("Load Wallet"),
+      //: Describing what is being done in background.
+      tr("Loading wallets…"));
```
Please fix indentation.

nit: s/"Load Wallet"/"Load Wallets"/ (singular -> plural).

jarolrod replied:
@shaavan apply clang format:

```diff
@@ -350,10 +350,10 @@ LoadWalletsActivity::LoadWalletsActivity(WalletController* wallet_controller, QW
 void LoadWalletsActivity::load()
 {
     showProgressDialog(
-      //: Title of progress window when which is displayed when wallets are being loaded.
-      tr("Load Wallet"),
-      //: Describing what is being done in background.
-      tr("Loading wallets…"));
+        //: Title of progress window when which is displayed when wallets are being loaded.
+        tr("Load Wallet"),
+        //: Describing what is being done in background.
+        tr("Loading wallets…"));
 
     QTimer::singleShot(0, worker(), [this] {
         for (auto& wallet : node().walletClient().getWallets()) {
```

shaavan replied:
> nit: s/"Load Wallet"/"Load Wallets"/ (singular -> plural).

Fixed

> apply clang-format:

Applied

Addressed in ccecfa8

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -231,7 +232,11 @@ void CreateWalletActivity::askPassphrase()
 {
-    showProgressDialog(tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
+    showProgressDialog(
+        //: Title of progress window when Create Wallet action is selected.
```
```suggestion
        //: Title of window indicating the progress of the creation of a new wallet.
```

shaavan replied:
Addressed in ccecfa8

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -231,7 +232,11 @@ void CreateWalletActivity::askPassphrase()
+        //: Title of progress window when Create Wallet action is selected.
+        tr("Create Wallet"),
+        //: Describing what is being done in background.
+        tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
```
while here, can we break up the HTML tags from the string to be translated? Keeping these tags as part of the string is error-prone

```diff
@@ -236,7 +236,7 @@ void CreateWalletActivity::createWallet()
         //: Title of progress window when Create Wallet action is selected.
         tr("Create Wallet"),
         //: Describing what is being done in background.
-        tr("Creating Wallet <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
+        tr("Creating Wallet") + QString(" <b>%1</b>…").arg(m_create_wallet_dialog->walletName().toHtmlEscaped()));
 
     std::string name = m_create_wallet_dialog->walletName().toStdString();
     uint64_t flags = 0;
@@ -331,7 +331,7 @@ void OpenWalletActivity::open(const std::string& path)
         //: Title of progress window when Open Wallet action is selected.
         tr("Open Wallet"),
         //: Describing what is being done in the background.
-        tr("Opening Wallet <b>%1</b>…").arg(name.toHtmlEscaped()));
+        tr("Opening Wallet") + QString(" <b>%1</b>…").arg(name.toHtmlEscaped()));
 
     QTimer::singleShot(0, worker(), [this, path] {
         std::unique_ptr<interfaces::Wallet> wallet = node().walletClient().loadWallet(path, m_error_message, m_warning_message);
```

shaavan replied:
> Keeping these tags as part of the string is error-prone

Oh, I was not aware of it. Thanks for enlightening me! Addressed in ccecfa8

hebasto replied:
> while here, can we break up the HTML tags from the string to be translated? Keeping these tags as part of the string is error-prone

Why?

![Screenshot from 2021-10-04 22-56-56](https://user-images.githubusercontent.com/32963518/135916075-502c09bc-32c6-432b-aece-8b120cc6054a.png)

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -231,7 +232,11 @@ void CreateWalletActivity::askPassphrase()
+    showProgressDialog(
+        //: Title of progress window when Create Wallet action is selected.
+        tr("Create Wallet"),
+        //: Describing what is being done in background.
```
```suggestion
        //: Descriptive text of the create wallet progress window which indicates to the user what wallet is currently being created.
```

shaavan replied:
This translator comment description looks a lot better! Addressed in ccecfa8

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -322,7 +327,11 @@ void OpenWalletActivity::open(const std::string& path)
 
-    showProgressDialog(tr("Opening Wallet <b>%1</b>…").arg(name.toHtmlEscaped()));
+    showProgressDialog(
+        //: Title of progress window when Open Wallet action is selected.
```
```suggestion
        //: Title of window indicating the progress of opening a wallet.
```

shaavan replied:
Addressed in ccecfa8

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -340,7 +349,11 @@ LoadWalletsActivity::LoadWalletsActivity(WalletController* wallet_controller, QW
-    showProgressDialog(tr("Loading wallets…"));
+    showProgressDialog(
+      //: Title of progress window when which is displayed when wallets are being loaded.
+      tr("Load Wallet"),
```
as already mentioned by @hebasto [here](https://github.com/bitcoin-core/gui/pull/409/files#r720722285)
```suggestion
      tr("Load Wallets"),
```

shaavan replied:
Addressed in ccecfa8

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -322,7 +327,11 @@ void OpenWalletActivity::open(const std::string& path)
+    showProgressDialog(
+        //: Title of progress window when Open Wallet action is selected.
+        tr("Open Wallet"),
+        //: Describing what is being done in the background.
```
```suggestion
        //: Descriptive text of the open wallet progress window which indicates to the user what wallet is currently being opened.
```

shaavan replied:
Addressed in ccecfa8

jarolrod commented on src/qt/walletcontroller.cpp:
```diff
@@ -340,7 +349,11 @@ LoadWalletsActivity::LoadWalletsActivity(WalletController* wallet_controller, QW
+    showProgressDialog(
+      //: Title of progress window when which is displayed when wallets are being loaded.
+      tr("Load Wallet"),
+      //: Describing what is being done in background.
```
```suggestion
      //: Descriptive text of the load wallets progress window which indicates to the user that wallets are currently being loaded.
```

shaavan replied:
Addressed in ccecfa8

shaavan replied:
Oh! Thanks for catching it! Addressed in 01bff8f

shaavan commented:
Updated from a086a37 to ccecfa86a010704ac5247fc637de28e71b3e71b2 ([pr409.07](https://github.com/shaavan/gui/tree/pr409.07) -> [pr409.08](https://github.com/shaavan/gui/tree/pr409.08))

Addressed @hebasto and @jarolrod comments

Changes:
1. `Load Wallet` -> `Load Wallets`
2. Clang format corrected
3. Separated HTML tags from strings to avoid the risk of error
4. Translator comments have been expanded upon and are now highly descriptive and clear thanks to @jarolrod suggestions
5. Updated commit message to account for changes in PR.

luke-jr commented:
I agree [with hebasto](https://github.com/bitcoin-core/gui/pull/409#discussion_r721674667). The strings shouldn't be broken up. Some languages might need to reorder the sentence.

shaavan commented:
Updated from ccecfa8 to 9de84f6 ([pr409.08](https://github.com/shaavan/gui/tree/pr409.08) -> [pr409.09](https://github.com/shaavan/gui/tree/pr409.09))

Addressed @hebasto and @luke-jr suggestions

Changes:
1. Rejoined the separated HTML tags with the strings.

hebasto commented on src/qt/walletcontroller.cpp:
```diff
@@ -340,7 +351,11 @@ LoadWalletsActivity::LoadWalletsActivity(WalletController* wallet_controller, QW
 {
-    showProgressDialog(tr("Loading wallets…"));
+    showProgressDialog(
+        //: Title of progress window when which is displayed when wallets are being loaded.
```
Just to be sure, is the first "when" correct?

```suggestion
        //: Title of progress window which is displayed when wallets are being loaded.
```
?

shaavan replied:
Oops! It shouldn't have been there. Let me quickly correct it!

shaavan replied:
Done Corrected! f86fe19

hebasto reviewed (approved):
ACK 9de84f66bede3aeb7974e0b8840370ba9e5fd0e3

shaavan commented:
Updated from 9de84f6 to f86fe19 ([pr409.09](https://github.com/shaavan/gui/tree/pr409.09) -> [pr409.10](https://github.com/shaavan/gui/tree/pr409.10))

Addressed @hebasto suggestion

Changes:
1. Corrected a grammatical mistake in the translator's comments.

luke-jr reviewed (approved):
utACK f86fe193329e057b7c6b624c3b3f11d5ae04826f

shaavan commented:
Updated from f86fe19 to 01bff8f ([pr409.10](https://github.com/shaavan/gui/tree/pr409.10) -> [pr409.11](https://github.com/shaavan/gui/tree/pr409.11))

Addressed @jarolrod’s suggestion

Changes:
1. Added the translator’s comment suggestion which was not pushed earlier in the PR.

hebasto reviewed (approved):
ACK 01bff8f0494ba1c02c087b7186c892785f326fd9, tested on Linux Mint 20.2 (Qt 5.12.8).

shaavan commented:
Thank you very much to everyone who took out time to review this PR. Especially, thanks to @jarolrod and @hebasto for sticking out till last in reviewing this PR and helping in improving it.
